hibernate-rx-core/src/main/java/org/hibernate/rx/impl/RxSessionInternalImpl.java

```diff
@@ -88,12 +123,46 @@ public Object immediateLoad(String entityName, Serializable id) throws Hibernate
 					} );
 		}
 		if ( association instanceof PersistentCollection ) {
-			//TODO: handle PersistentCollection (raise InitializeCollectionEvent)
-			throw new UnsupportedOperationException("fetch() is not yet implemented for collections");
+			PersistentCollection persistentCollection = (PersistentCollection) association;
+			return rxInitializeCollection( persistentCollection, false )
+					.thenApply( l -> Optional.ofNullable( (T) l ) );
```

Comment:
@DavideD this can be written more cleanly as `.thenApply(Optional::ofNullable))`, I think.

But no matter, I will make that change later if you like.

Reply:
I'm still not used to this kind of syntax :)

Reply:
Fixed in 91446b6fcc4893ed3b152f60aaef0acebd47d825